### PR TITLE
JAVA-2332: Destroy connection pool when a node gets removed

### PR DESCRIPTION
**Pull request by olim7t**

### CODE REVIEW

File: core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java [hunk 1]
```diff
@@ -314,10 +314,14 @@ private void processStateEvent(NodeStateEvent event) {
...
-      } else if (newState == NodeState.FORCED_DOWN) {
+      } else if (newState == null || newState == NodeState.FORCED_DOWN) {
         ChannelPool pool = pools.remove(node);
         if (pool != null) {
-          LOG.debug("[{}] {} was FORCED_DOWN, destroying pool", logPrefix, node);
+          LOG.debug(
+              "[{}] {} was {}, destroying pool",
+              logPrefix,
+              node,
+              newState == null ? "removed" : "forced down");
```
Comment (emerkle826) [inline core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java]: total nit here: Does it matter much that the debug log changes `FORCED_DOWN` to `forced down` in those cases? I realize it's just a log, and a Debug log at that. Again, total nitpick.

**Review by emerkle826 — APPROVED** [review]
LGTM :+1:

File: core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java (hunk 1 above)
Comment (olim7t) [reply core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java]: You're right, debug logs don't matter much but there's no reason to change it either. If anything, using the constant name gives you a hint of what to look for in the source code.
I'll change it back.

File: core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java [hunk 2]
```diff
@@ -314,10 +314,14 @@ private void processStateEvent(NodeStateEvent event) {
       NodeState newState = event.newState;
       if (pending.containsKey(node)) {
         pendingStateEvents.put(node, event);
-      } else if (newState == NodeState.FORCED_DOWN) {
+      } else if (newState == null || newState == NodeState.FORCED_DOWN) {
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java:317]: So when the state is `null` that means the node has been removed? Can we document this somewhere?

Comment (olim7t) [reply core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java:317]: Yes, `NodeStateEvent` is a bit bare-bones: added is (null, UNKNOWN) and removed is (null, null). I'll add javadocs to the fields.

**Pull request merged**